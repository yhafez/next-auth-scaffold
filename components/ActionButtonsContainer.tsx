// Path: ./components/ActionButtonsContainer.tsx
import { ReactNode } from 'react'
import { Box } from '@mui/material'

export interface ActionButtonsContainerProps {
	name: string
	children: ReactNode
}

export default function ActionButtonsContainer({ name, children }: ActionButtonsContainerProps) {
	return (
		<Box
			id={`${name}-buttons-container`}
			sx={{
				display: 'flex',
				flexDirection: 'row',
				justifyContent: 'space-around',
				alignItems: 'center',
				gap: 2,
				margin: '8px 4px',
				width: '80%',
			}}
		>
			{children}
		</Box>
	)
}
