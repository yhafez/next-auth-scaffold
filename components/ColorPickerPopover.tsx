// Path: ./components/ColorPickerPopover.tsx

import { useState, useCallback } from 'react'
import { Popover, PopoverOrigin } from '@mui/material'
import { useBoundStore } from '../store'
import { ChromePicker } from 'react-color'

export interface ColorPickerPopoverProps {
	name: string
	open: boolean
	anchorEl: HTMLButtonElement | null
	handleClose: () => void
	handleColorChange: (color: string) => void
	anchorOrigin?: PopoverOrigin
	transformOrigin?: PopoverOrigin
}

const ColorPickerPopover = ({
	name,
	open,
	anchorEl,
	handleClose,
	handleColorChange,
	anchorOrigin,
	transformOrigin,
}: ColorPickerPopoverProps) => {
	const { customPalette } = useBoundStore()

	const [color, setColor] = useState(customPalette.primary.main)

	const handleChange = useCallback(
		(color: any) => {
			setColor(color.hex)
			handleColorChange(color.hex)
		},
		[handleColorChange],
	)

	return (
		<Popover
			id={`${name}-color-picker-popover`}
			open={open}
			anchorEl={anchorEl}
			onClose={handleClose}
			anchorOrigin={anchorOrigin}
			transformOrigin={transformOrigin}
			sx={{
				cursor: 'pointer',
				'& .MuiPopover-paper': {
					padding: 2,
					backgroundColor: 'background.paper',
					color: 'text.primary',
				},
			}}
		>
			<ChromePicker color={color} onChange={color => handleChange(color)} disableAlpha />
		</Popover>
	)
}

export default ColorPickerPopover
