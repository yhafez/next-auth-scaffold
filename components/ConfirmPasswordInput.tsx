// Path: ./components/ConfirmPasswordInput.tsx
import { Dispatch, SetStateAction, KeyboardEvent } from 'react'

import PasswordInput from './PasswordInput'

export interface ConfirmPasswordInputProps {
	name: string
	password: string
	setPassword: Dispatch<SetStateAction<string>>
	confirmPassword: string
	setConfirmPassword: Dispatch<SetStateAction<string>>
	disabled?: boolean
	handleEnter?: (e: KeyboardEvent) => void
}

export default function ConfirmPasswordInput({
	name,
	password,
	setPassword,
	confirmPassword,
	setConfirmPassword,
	disabled = false,
	handleEnter,
}: ConfirmPasswordInputProps) {
	return (
		<>
			<PasswordInput
				name={name}
				value={password}
				setValue={setPassword}
				disabled={disabled}
				handleEnter={handleEnter}
			/>
			<PasswordInput
				name={`${name}`}
				value={confirmPassword}
				setValue={setConfirmPassword}
				disabled={disabled}
				isConfirmPassword={true}
				handleEnter={handleEnter}
			/>
		</>
	)
}
