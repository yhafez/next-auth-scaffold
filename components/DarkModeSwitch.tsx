// Path: ./components/DarkModeSwitch.tsx
import { useEffect, useState } from 'react'
import { Box, Switch, Typography } from '@mui/material'
import WbSunnyIcon from '@mui/icons-material/WbSunny'
import Brightness2Icon from '@mui/icons-material/Brightness2'

import { useBoundStore } from '../store'

export interface DarkModeSwitchProps {
	name: string
}

export default function DarkModeSwitch({ name }: DarkModeSwitchProps) {
	const { darkMode, setDarkMode } = useBoundStore()
	const [selected, setSelected] = useState(false)

	const handleDarkMode = () => {
		localStorage.setItem('darkMode', JSON.stringify(!darkMode))
		setDarkMode(!darkMode)
	}

	useEffect(() => {
		const darkMode = JSON.parse(localStorage.getItem('darkMode') || 'false')
		setDarkMode(darkMode)
	}, [darkMode, setDarkMode])

	const handleEnter = (e: React.KeyboardEvent<HTMLButtonElement>) => {
		if (e.key === 'Enter') {
			handleDarkMode()
		}
	}

	return (
		<Box
			id={`${name}-dark-mode-switch-container`}
			onMouseEnter={() => setSelected(true)}
			onMouseLeave={() => setSelected(false)}
			sx={{
				display: 'flex',
				flexDirection: 'column',
				justifyContent: 'space-around',
				alignItems: 'center',
				height: '100%',
				mt: 1,
			}}
		>
			<Switch
				id={`${name}-dark-mode-switch`}
				checked={darkMode}
				onChange={handleDarkMode}
				onKeyDown={handleEnter}
				aria-labelledby={`${name}-dark-mode-switch-label-${darkMode ? 'dark' : 'light'}`}
				sx={{
					color: selected ? (darkMode ? 'primary.light' : 'primary.dark') : 'primary.contrastText',
					'& .MuiSwitch-switchBase': {
						color: 'primary.contrastText',
						'&.Mui-checked': {
							color: 'primary.contrastText',
						},
						'&.Mui-checked + .MuiSwitch-track': {
							backgroundColor: 'secondary.main',
						},
					},
				}}
				checkedIcon={
					<Brightness2Icon
						id={`${name}-dark-mode-switch-icon-checked`}
						sx={{
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					/>
				}
				icon={
					<WbSunnyIcon
						id={`${name}-dark-mode-switch-icon-unchecked`}
						sx={{
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					/>
				}
			/>
			<label htmlFor={`${name}-dark-mode-switch`}>
				<Typography
					id={`${name}-dark-mode-switch-label-${darkMode ? 'dark' : 'light'}`}
					variant="body2"
					sx={{
						fontWeight: 500,
						color: selected
							? darkMode
								? 'primary.light'
								: 'primary.dark'
							: 'primary.contrastText',
						cursor: 'pointer',
					}}
				>
					{darkMode ? 'Dark Mode' : 'Light Mode'}
				</Typography>
			</label>
		</Box>
	)
}
