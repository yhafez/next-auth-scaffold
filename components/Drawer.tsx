// Path: ./components/Drawer.tsx
import { Drawer, IconButton, Box, Typography, useMediaQuery } from '@mui/material'
import { ChevronLeft } from '@mui/icons-material'

import { useBoundStore } from '../store'

export interface DrawerProps {
	open: boolean
	handleDrawerClose: () => void
	children: React.ReactNode
}

export default function DrawerComponent({ open, handleDrawerClose, children }: DrawerProps) {
	const { darkMode, theme } = useBoundStore()
	const isMobile = useMediaQuery(`(max-width:${theme.breakpoints.values.sm}px)`)

	return (
		<Drawer
			id="drawer"
			sx={{
				width: '18em',
				flexShrink: 0,
				overflowY: 'scroll',
				'& .MuiDrawer-paper': {
					width: '18em',
					boxSizing: 'border-box',
					backgroundColor: darkMode ? 'primary.dark' : 'primary.light',
					color: 'primary.contrastText',
					display: 'flex',
					flexDirection: 'column',
					justifyContent: 'flex-start',
				},
			}}
			variant={isMobile ? 'temporary' : 'permanent'}
			open={open}
			anchor="left"
			onClose={handleDrawerClose}
		>
			<>
				{isMobile && (
					<Box
						id="drawer-close-button-container"
						onClick={handleDrawerClose}
						sx={{
							display: 'flex',
							alignItems: 'center',
							justifyContent: 'space-between',
							padding: 1,
							...theme.mixins.toolbar,
							borderBottom: '1px solid rgba(255, 255, 255, 0.12)',
							pt: 2,
							pb: 2,
							px: 0,
							width: '100%',
							height: '2em',
							cursor: 'pointer',
							'&:hover': {
								backgroundColor: 'primary.main',
							},
							'&:focus': {
								backgroundColor: 'primary.main',
							},
							'&:active': {
								backgroundColor: 'primary.main',
							},
						}}
					>
						<IconButton
							id="drawer-close-button"
							sx={{
								color: 'primary.contrastText',
								height: '100%',
								borderRadius: 0,
								'&:hover': {
									backgroundColor: 'transparent',
								},
							}}
						>
							<ChevronLeft
								id="drawer-close-button-icon"
								sx={{ fontSize: 40 }}
								role="img"
								aria-labelledby="drawer-close-button-label-text"
							/>
						</IconButton>
						<label id="drawer-close-button-label" htmlFor="drawer-close-button">
							<Typography
								id="drawer-close-button-label-text"
								sx={{
									fontWeight: 500,
									pr: 2,
									color: 'primary.contrastText',
									ml: 1,
									cursor: 'pointer',
									'&:hover': {
										color: darkMode ? 'primary.light' : 'primary.dark',
										transform: 'scale(1.1)',
									},
									'&:focus': {
										color: darkMode ? 'primary.light' : 'primary.dark',
										transform: 'scale(1.1)',
									},
									'&:active': {
										color: darkMode ? 'primary.light' : 'primary.dark',
										transform: 'scale(1.1)',
									},
								}}
							>
								Close
							</Typography>
						</label>
					</Box>
				)}
				{children}
			</>
		</Drawer>
	)
}
