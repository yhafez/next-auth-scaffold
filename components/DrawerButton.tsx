// Path: ./components/DrawerButton.tsx
import { ReactNode, MouseEvent } from 'react'
import Link from 'next/link'
import { useRouter } from 'next/router'
import { Link as MuiLink, IconButton, Typography } from '@mui/material'

export interface DrawerButtonProps {
	name: string
	label: string
	icon: ReactNode
	handleClick?: (e: MouseEvent<HTMLButtonElement>) => void
	reverse?: boolean
	href?: string
}

export default function DrawerButton({
	name,
	label,
	icon,
	handleClick,
	reverse,
	href,
}: DrawerButtonProps) {
	const router = useRouter()
	const isSelected = router.pathname === href ? true : false

	if (href) {
		return (
			<MuiLink
				id={`${name}-drawer-link`}
				component={Link}
				href={href}
				aria-current={isSelected ? 'page' : undefined}
				title={label}
				aria-labelledby={`${name}-drawer-button-label`}
				sx={{
					textDecoration: 'none',
					color: 'primary.contrastText',
					padding: 0,
					display: 'flex',
					alignItems: 'center',
					justifyContent: 'space-between',
					width: '100%',
					borderRadius: 0,
					backgroundColor: isSelected ? 'primary.main' : 'transparent',
					p: 4,
					'&:hover': {
						backgroundColor: 'primary.main',
					},
					'&:focus': {
						backgroundColor: 'primary.main',
					},
					'&:active': {
						backgroundColor: 'primary.main',
					},
				}}
			>
				{reverse && icon}
				<Typography
					id={`${name}-drawer-button-label`}
					variant="body1"
					sx={{
						fontSize: '1.2rem',
						fontWeight: 'bold',
					}}
				>
					{label}
				</Typography>
				{!reverse && icon}
			</MuiLink>
		)
	}

	return (
		<IconButton
			id={`${name}-drawer-button`}
			onClick={handleClick}
			disableRipple
			sx={{
				color: 'primary.contrastText',
				padding: 0,
				display: 'flex',
				alignItems: 'center',
				justifyContent: 'space-between',
				width: '100%',
				borderRadius: 0,
				backgroundColor: isSelected ? 'primary.main' : 'transparent',
				p: 4,
				'&:hover': {
					backgroundColor: 'primary.main',
				},
				'&:focus': {
					backgroundColor: 'primary.main',
				},
				'&:active': {
					backgroundColor: 'primary.main',
				},
			}}
		>
			{reverse && icon}
			<Typography
				id={`${name}-drawer-button-label`}
				variant="body1"
				sx={{
					fontSize: '1.2rem',
					fontWeight: 'bold',
				}}
			>
				{label}
			</Typography>
			{!reverse && icon}
		</IconButton>
	)
}
