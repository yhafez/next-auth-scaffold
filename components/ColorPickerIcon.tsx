// Path: ./components/ColorPicker.tsx
import { useState, MouseEvent, useCallback, useEffect } from 'react'
import { Box, IconButton, Typography } from '@mui/material'
import ColorLensIcon from '@mui/icons-material/ColorLens'
import { darken, lighten } from 'color2k'

import ColorPickerPopover from './ColorPickerPopover'

import { useBoundStore } from '../store'
import { getContrastColor, getSecondaryColor } from '../utils/helpers'

export interface ColorPickerProps {
	name: string
}

export default function ColorPickerIcon({ name }: ColorPickerProps) {
	const { darkMode, customPalette, setCustomPalette } = useBoundStore()

	const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null)
	const [selected, setSelected] = useState(false)

	const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
		setAnchorEl(event.currentTarget)
	}

	const handleClose = () => {
		setAnchorEl(null)
	}

	const open = Boolean(anchorEl)
	const id = open ? 'simple-popover' : undefined

	const handleColorChange = useCallback(
		(color: string) => {
			localStorage.setItem('customPalette', color)

			const contrast = getContrastColor(color)
			const secondaryColor = getSecondaryColor(color)
			const secondaryColorContrast = getContrastColor(secondaryColor)

			const newPalette = {
				...customPalette,
				primary: {
					main: color,
					contrastText: contrast,
					light: lighten(color, 0.2),
					dark: darken(color, 0.2),
				},
				secondary: {
					main: secondaryColor,
					contrastText: secondaryColorContrast,
					light: lighten(secondaryColor, 0.2),
					dark: darken(secondaryColor, 0.2),
				},
			}

			setCustomPalette(newPalette)
		},
		[customPalette, setCustomPalette],
	)

	useEffect(() => {
		const customPalette = localStorage.getItem('customPalette')

		if (customPalette) {
			handleColorChange(customPalette)
		}
	}, [])

	return (
		<Box
			id={`${name}-color-picker-container`}
			onMouseEnter={() => setSelected(true)}
			onMouseLeave={() => setSelected(false)}
			sx={{
				mr: 4,
				height: '100%',
				display: 'flex',
				alignItems: 'center',
				justifyContent: 'center',
			}}
		>
			<Box
				id={`${name}-color-picker-box`}
				sx={{
					display: 'flex',
					flexDirection: 'column',
					alignItems: 'center',
					justifyContent: 'center',
					mt: 2,
					p: 0,
					height: '100%',
					cursor: 'pointer',
				}}
			>
				<IconButton
					id={`${name}-color-picker-button`}
					aria-describedby={id}
					aria-labelledby={`${name}-color-picker-label`}
					onClick={handleClick}
				>
					<ColorLensIcon
						id={`${name}-color-picker-icon`}
						sx={{
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
							fontSize: '1.5rem',
						}}
					/>
				</IconButton>
				<label htmlFor={`${name}-color-picker-button`}>
					<Typography
						id={`${name}-color-picker-label`}
						variant="body2"
						sx={{
							fontWeight: 500,
							cursor: 'pointer',
							color: selected
								? darkMode
									? 'primary.light'
									: 'primary.dark'
								: 'primary.contrastText',
						}}
					>
						Theme Color
					</Typography>
				</label>
			</Box>
			<ColorPickerPopover
				name={name}
				open={open}
				anchorEl={anchorEl}
				handleClose={handleClose}
				anchorOrigin={{
					vertical: 'bottom',
					horizontal: 'center',
				}}
				transformOrigin={{
					vertical: 'top',
					horizontal: 'center',
				}}
				handleColorChange={handleColorChange}
			/>
		</Box>
	)
}
