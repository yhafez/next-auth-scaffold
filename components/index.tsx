// Path: ./components/index.tsx
import Layout from './Layout'

import DashboardDrawer from './Drawers/DashboardDrawer'
import SettingsDrawer from './Drawers/SettingsDrawer'

import ColorPickerPopover from './Popovers/ColorPickerPopover'
import DrawerSignOutPopover from './Popovers/DrawerSignOutPopover'

import ActionButtonsContainer from './ActionButtonsContainer'
import ColorPickerIcon from './ColorPickerIcon'
import ConfirmPasswordInput from './ConfirmPasswordInput'
import DarkModeSwitch from './DarkModeSwitch'
import Drawer from './Drawer'
import DrawerButton from './DrawerButton'
import EditableTextField from './EditableTextField'
import EmailInput from './EmailInput'
import Modal from './Modal'
import ModalButton from './ModalButton'
import ModalNote from './ModalNote'
import NavigationButton from './NavigationButton'
import PasswordInput from './PasswordInput'
import SubmitButton from './SubmitButton'
import VisibilityIconButton from './VisibilityIconButton'

export {
	Layout,
	DashboardDrawer,
	SettingsDrawer,
	ColorPickerPopover,
	DrawerSignOutPopover,
	ActionButtonsContainer,
	ColorPickerIcon,
	ConfirmPasswordInput,
	DarkModeSwitch,
	Drawer,
	DrawerButton,
	EditableTextField,
	EmailInput,
	Modal,
	ModalButton,
	ModalNote,
	NavigationButton,
	PasswordInput,
	SubmitButton,
	VisibilityIconButton,
}
