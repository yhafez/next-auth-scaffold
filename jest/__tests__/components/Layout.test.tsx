// Path: ./jest/__tests__/components/Layout.test.tsx
import { render, act, screen, waitFor } from '@testing-library/react'
import { axe } from 'jest-axe'
import { SessionProvider } from 'next-auth/react'

import { Layout } from '../../../components'

describe('Layout', () => {
	let container: HTMLElement

	beforeEach(() => {
		act(() => {
			container = render(
				<SessionProvider
					session={{
						user: {
							name: 'test',
							email: 'test@example.com',
							image: 'test',
						},
						expires: '1d',
					}}
				>
					<Layout name="test" pageTitle="test">
						<div id="children">Children</div>
					</Layout>
				</SessionProvider>,
			).container

			return container
		})
	})

	it('Should have no accessibility violations', async () => {
		waitFor(async () => expect(await axe(screen.getByRole('main'))).toHaveNoViolations())
	})

	it('Should have a Box element with id "test-layout-container"', () => {
		const test = container.querySelector('#test-layout-container')
		expect(test).toBeTruthy()
	})

	it('Should have an Appbar element with id "test-app-bar"', () => {
		const test = container.querySelector('#test-app-bar')
		expect(test).toBeTruthy()
	})

	it('Should have a Toolbar element with id "test-toolbar"', () => {
		const test = container.querySelector('#test-toolbar')
		expect(test).toBeTruthy()
	})

	it('Should have a Box element with id "test-toolbar-title"', () => {
		const test = container.querySelector('#test-toolbar-title')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with text "Next Auth Scaffold"', () => {
		const test = container.querySelector('#test-toolbar-title')?.textContent
		expect(test).toBe('Next Auth Scaffold')
	})

	it('Should have a Box element with id "test-layout-content"', () => {
		const test = container.querySelector('#test-layout-content')
		expect(test).toBeTruthy()
	})

	it('Should have a Typography element with text "Children"', () => {
		const test = container.querySelector('#children')
		expect(test?.textContent).toBe('Children')
	})

	it('Should match snapshot', () => {
		expect(container).toMatchSnapshot()
	})
})
