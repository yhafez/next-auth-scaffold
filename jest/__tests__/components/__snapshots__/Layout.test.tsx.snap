// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-oqgvqi"
    id="test-layout-container"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1w9mh9n-MuiPaper-root-MuiAppBar-root"
      id="test-app-bar"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-u4fbkj-MuiToolbar-root"
        id="test-toolbar"
      >
        <h1
          class="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-11cao63-MuiTypography-root"
          id="test-toolbar-title"
        >
          Next Auth Scaffold
        </h1>
        <div
          class="MuiBox-root css-zd0i21"
          id="test-toolbar-color-picker-container"
        >
          <div
            class="MuiBox-root css-1rks40w"
            id="test-toolbar-color-picker-box"
          >
            <button
              aria-labelledby="test-toolbar-color-picker-label"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              id="test-toolbar-color-picker-button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1kqeszy-MuiSvgIcon-root"
                data-testid="ColorLensIcon"
                focusable="false"
                id="test-toolbar-color-picker-icon"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <label
              for="test-toolbar-color-picker-button"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-28mjxw-MuiTypography-root"
                id="test-toolbar-color-picker-label"
              >
                Theme Color
              </p>
            </label>
          </div>
        </div>
        <div
          class="MuiBox-root css-1lvwh4e"
          id="test-toolbar-dark-mode-switch-container"
        >
          <span
            class="MuiSwitch-root MuiSwitch-sizeMedium css-14s0ud6-MuiSwitch-root"
          >
            <span
              aria-labelledby="test-toolbar-dark-mode-switch-label-light"
              class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
            >
              <input
                class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                id="test-toolbar-dark-mode-switch"
                type="checkbox"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
                data-testid="WbSunnyIcon"
                focusable="false"
                id="test-toolbar-dark-mode-switch-icon-unchecked"
                viewBox="0 0 24 24"
              >
                <path
                  d="m6.76 4.84-1.8-1.79-1.41 1.41 1.79 1.79 1.42-1.41zM4 10.5H1v2h3v-2zm9-9.95h-2V3.5h2V.55zm7.45 3.91-1.41-1.41-1.79 1.79 1.41 1.41 1.79-1.79zm-3.21 13.7 1.79 1.8 1.41-1.41-1.8-1.79-1.4 1.4zM20 10.5v2h3v-2h-3zm-8-5c-3.31 0-6 2.69-6 6s2.69 6 6 6 6-2.69 6-6-2.69-6-6-6zm-1 16.95h2V19.5h-2v2.95zm-7.45-3.91 1.41 1.41 1.79-1.8-1.41-1.41-1.79 1.8z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
            />
          </span>
          <label
            for="test-toolbar-dark-mode-switch"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-28mjxw-MuiTypography-root"
              id="test-toolbar-dark-mode-switch-label-light"
            >
              Light Mode
            </p>
          </label>
        </div>
      </div>
    </header>
    <div
      class="MuiBox-root css-1i89r39"
      id="test-layout-content"
    >
      <div>
        Children
      </div>
    </div>
  </div>
</div>
`;
